COLLABORATOR_GUIDE.md

```diff
@@ -169,11 +169,12 @@ the comment anyway to avoid any doubt.
 
 ### Testing and CI
 
-All fixes must have a test case which demonstrates the defect. The test should
+All fixes should have a test case which demonstrates the defect. The test should
```

Comment:
I changed this because we sometimes don't require test requests when tests are impractical to write.

Reply:
Using "should" conveys a lower-grade requirement. I think using "must" but noting that the collaborator can ask for an exception is better. I would not want to land a PR that did not have a test - I think lowering this requirement would get our codebase worse.

Reply:
@mcollina Since it's a only-somewhat-related change to the main change in this PR, I'll put it back to what it was.